javagen/src/main/java/com/azure/autorest/model/clientmodel/ArrayType.java

```diff
@@ -11,26 +11,28 @@
  * The details of an array type that is used by a client.
  */
 public class ArrayType implements IType {
-    public static final ArrayType ByteArray = new ArrayType(PrimitiveType.Byte, (String defaultValueExpression) -> defaultValueExpression == null ? "new byte[0]" : String.format("\"%1$s\".getBytes()", defaultValueExpression));
-    private IType elementType;
-    private Function<String, String> defaultValueExpressionConverter;
+    public static final ArrayType ByteArray = new ArrayType(PrimitiveType.Byte,
+        defaultValueExpression -> defaultValueExpression == null
+            ? "new byte[0]" // TODO (alzimmer): Should this be new byte[0] or null?
```

Comment:
if I understand the problem correctly, I think the default  should be `null` so there is a way to represent an empty array and distinguish it from lack of value.

Reply:
I'm going to file a follow-up issue for this